### PR TITLE
[KHWEBA - 46]Main content,drawer

### PR DESCRIPTION
**Pull request by OlgaOrlova**
image

### CODE REVIEW

**Review by Vladyslav-Plakhuta — DISMISSED** [review]

File: index.html [hunk 1]
```diff
@@ -5,6 +5,10 @@
   <meta charset="UTF-8">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
+  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700,400italic">
+  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```
Comment (gift-a) [inline index.html]: Why do you add material icons stylesheet twice?

File: src/components/button-more/button-more.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import angular from "angular";
+import template from "./button-more.html";
+class ButtonMoreController {}
+export const buttonMoreModule = angular
+  .module("buttonMore", ["ngMaterial"])
```
Comment (gift-a) [inline src/components/button-more/button-more.js]: You need to import ngMaterial in every module which needs it

File: src/components/accounts/accounts.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        currency: "UAH",
+        id: "i78h0bobom"
+      }
+    ];
+  }
+
+  accountsItemClick() {}
+}
+export const accountsModule = angular
+  .module("accounts", ["ngMaterial", buttonMoreModule])
```
Comment (gift-a) [inline src/components/accounts/accounts.js]: you are using ngMaterial here, but haven't imported it.

File: src/components/app/app.js [hunk 4]
```diff
@@ -1,6 +1,29 @@
 import angular from "angular";
 import template from "./app.html";
+import { drawerModule } from "../drawer/drawer";
+import "angular-animate";
+import "angular-aria";
+import "angular-material";
```
Comment (gift-a) [inline src/components/app/app.js]: it seems like you don't use material in this component

File: src/components/accounts/accounts.html [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<md-list class="accounts drawer-menu" ng-cloak>
+  <md-list-item ng-repeat="item in $ctrl.accounts" ng-click="accountsItemClick()">
+    <md-icon md-font-set="material-icons"> account_balance </md-icon>
+    <p>
+      {{item.name}} {{item.balance}} {{item.currency}}
```
Comment (ihor7016) [inline src/components/accounts/accounts.html]: please remove space between balance and currency (like in vanilla app)

File: src/components/accounts/accounts.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import angular from "angular";
+import template from "./accounts.html";
+import { buttonMoreModule } from "../button-more/button-more";
+class AccountsController {
```
Comment (ihor7016) [inline src/components/accounts/accounts.js]: please separate imports with new line

File: src/components/accounts/accounts.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        name: "reretr",
+        balance: 2222,
+        type: "checking",
+        currency: "UAH",
+        id: "i78h0bobom"
+      }
+    ];
+  }
+
+  accountsItemClick() {}
```
Comment (ihor7016) [inline src/components/accounts/accounts.js]: maybe `handleItemClick` ?

File: src/components/app/app.js [hunk 8]
```diff
@@ -1,6 +1,29 @@
...
+    this.tasks.push(task);
+  }
+
+  openDrawer() {
+    this.drawerOpened.value = !this.drawerOpened.value;
+  }
+}
+
+export const appModule = angular
+  .module("appModule", ["ngMaterial", drawerModule, toolbarModule])
```
Comment (ihor7016) [inline src/components/app/app.js]: please use imported ngMaterial here

File: src/components/button-more/button-more.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import angular from "angular";
+import "angular-animate";
+import "angular-aria";
+import "angular-material";
+import template from "./button-more.html";
+class ButtonMoreController {}
+export const buttonMoreModule = angular
```
Comment (ihor7016) [inline src/components/button-more/button-more.js]: add new lines between import class and export

File: src/components/button-more/button-more.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+  .module("buttonMore", ["ngMaterial"])
```
Comment (ihor7016) [inline src/components/button-more/button-more.js]: please use imported ngMaterial

File: src/components/drawer/drawer.html [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<md-sidenav class="drawer md-sidenav-left" md-is-locked-open="$ctrl.isOpened" md-disable-backdrop md-whiteframe="4">
+  <md-list-item ng-click="addAccountClick($event)">
```
Comment (ihor7016) [inline src/components/drawer/drawer.html]: `onAccountClick` or `handleAccountClick`

File: src/components/drawer/drawer.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+import template from "./drawer.html";
+import { accountsModule } from "../accounts/accounts";
+import { tagsModule } from "../tags/tags";
+
+export class DrawerController {
+  constructor($mdSidenav) {
+    this.$mdSidenav = $mdSidenav;
+    this.isOpened = false;
+  }
+  addAccountClick(event) {}
```
Comment (ihor7016) [inline src/components/drawer/drawer.js]: please separate methods with newline + add at least console log there

File: src/components/drawer/drawer.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+  toggleLeft() {
+    this.isOpened = !this.isOpened;
+  }
+}
+export const drawerModule = angular
+  .module("drawer", ["ngMaterial", accountsModule, tagsModule])
+  .component("drawer", {
+    controller: DrawerController,
+    bindings: {
+      drawerOpened: "<"
```
Comment (ihor7016) [inline src/components/drawer/drawer.js]: it is boolean variable so it should start with `is` or `has` -- `isDrawerOpened` or `isOpen`

File: src/components/tags/tags.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
+import angular from "angular";
+import "angular-animate";
+import "angular-aria";
+import "angular-material";
+import template from "./tags.html";
+import { buttonMoreModule } from "../button-more/button-more";
+class TagsController {
```
Comment (ihor7016) [inline src/components/tags/tags.js]: newline between import class and export

File: src/components/tags/tags.js [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+  constructor() {
+    this.tags = ["tag1", "tag2", "tag3"];
+  }
+
+  tagsItemClick() {}
```
Comment (ihor7016) [inline src/components/tags/tags.js]: `handle` + console.log

File: src/components/toolbar/toolbar.scss [hunk 16]
```diff
@@ -0,0 +1,3 @@
+.toolbardemoBasicUsage md-toolbar md-icon.md-default-theme {
```
Comment (ihor7016) [inline src/components/toolbar/toolbar.scss]: please make this classname follow convention. Where is BEM?

File: src/components/accounts/accounts.js [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+        balance: 2222,
+        type: "checking",
+        currency: "UAH",
+        id: "i78h0bobom"
+      }
+    ];
+  }
+
+  handleItemClick() {
+    console.log();
```
Comment (ihor7016) [inline src/components/accounts/accounts.js]: print something to screen -- so I will be able to test it

File: src/components/drawer/drawer.js [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+  constructor($mdSidenav) {
+    this.$mdSidenav = $mdSidenav;
+    this.isOpened = false;
+  }
+  handleAccountClick(event) {
+    console.log();
+  }
+
+  addTagClick(event) {
+    console.log();
```
Comment (ihor7016) [inline src/components/drawer/drawer.js]: why empty console.log here?

File: src/components/drawer/drawer.scss [hunk 19]
```diff
@@ -0,0 +1,3 @@
+.drawer {
+  height: 100vh;
```
Comment (ihor7016) [inline src/components/drawer/drawer.scss]: why not 100%?

File: src/components/tags/tags.js [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+import template from "./tags.html";
+import { buttonMoreModule } from "../button-more/button-more";
+
+class TagsController {
+  constructor() {
+    this.tags = ["tag1", "tag2", "tag3"];
+  }
+
+  handleItemClick() {
+    console.log();
```
Comment (ihor7016) [inline src/components/tags/tags.js]: why empty console.log

File: src/components/app/app.js [hunk 21]
```diff
@@ -1,6 +1,26 @@
...
+import { drawerModule } from "../drawer/drawer";
+import { toolbarModule } from "../toolbar/toolbar";
 
-export const appModule = angular.module("appModule", []).component("app", {
-  template
-}).name;
+export class AppController {
+  constructor() {
+    this.tasks = [];
+    this.drawerOpened = { value: false };
```
Comment (ihor7016) [inline src/components/app/app.js]: why this is an object?

Comment (ihor7016) [inline src/components/app/app.js]: you can do the same with just property: isOpen

File: src/components/button-more/button-more.js [hunk 22]
```diff
@@ -0,0 +1,14 @@
+import angular from "angular";
+import ngAnimate from "angular-animate";
+import ngAria from "angular-aria";
+import ngMaterial from "angular-material";
+import template from "./button-more.html";
+
+class ButtonMoreController {}
```
Comment (ihor7016) [inline src/components/button-more/button-more.js]: export class

File: src/components/toolbar/toolbar.scss [hunk 23]
```diff
@@ -0,0 +1,3 @@
+.toolbar md-toolbar md-icon.md-default-theme {
```
Comment (ihor7016) [inline src/components/toolbar/toolbar.scss]: .toolbar {
  & md-toolbar md-icon.md-default-theme { . .. }
}

File: src/components/drawer/drawer.js [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+  handleAccountClick(event) {
+    console.log("handleAccountClick");
+  }
+
+  addTagClick(event) {
+    console.log("addTagClick");
+  }
+
+  $onChanges(changes) {
+    this.toggleLeft();
```
Comment (ihor7016) [inline src/components/drawer/drawer.js]: you do not need this -- you can directly use `isDrawerOpened ` binding in template

File: src/components/toolbar/toolbar.scss [hunk 25]
```diff
@@ -0,0 +1,5 @@
+.toolbar {
+  & md-toolbar md-icon.md-default-theme {
```
Comment (ihor7016) [inline src/components/toolbar/toolbar.scss]: .md-toolbar .md-icon

File: src/components/toolbar/toolbar.html [hunk 26]
```diff
@@ -0,0 +1,12 @@
+<md-toolbar class="toolbar md-hue-2">
+  <div class="md-toolbar-tools">
+    <md-button class="md-icon-button" ng-click="$ctrl.handleMenuClick()" aria-label="Favorite">
+      <md-icon>
+        <i class="material-icons">menu</i>
+      </md-icon>
+    </md-button>
+    <h2 flex md-truncate>Buxy</h2>
+    <span flex=""></span>
```
Comment (ihor7016) [inline src/components/toolbar/toolbar.html]: why this prop is empty?

File: src/components/drawer/drawer.html [hunk 27]
```diff
@@ -0,0 +1,13 @@
+<md-sidenav class="drawer md-sidenav-left" md-is-locked-open="$ctrl.isOpened" md-disable-backdrop md-whiteframe="4">
+  <md-list-item ng-click="$ctrl.handleAccountClick($event)">
+    <md-icon md-font-set="material-icons"> add </md-icon>
+    <p> Add new account</p>
+  </md-list-item>
+  <accounts></accounts>
+  <md-divider></md-divider>
+  <md-list-item ng-click="$ctrl.addTagClick($event)">
```
Comment (ihor7016) [inline src/components/drawer/drawer.html]: this should be inside tags component

File: src/components/drawer/drawer.html [hunk 28]
```diff
@@ -0,0 +1,13 @@
+<md-sidenav class="drawer md-sidenav-left" md-is-locked-open="$ctrl.isOpened" md-disable-backdrop md-whiteframe="4">
+  <md-list-item ng-click="$ctrl.handleAccountClick($event)">
```
Comment (ihor7016) [inline src/components/drawer/drawer.html]: this  should be inside accounts component

File: src/components/accounts/accounts.js [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+        name: "reretr",
+        balance: 2222,
+        type: "checking",
+        currency: "UAH",
+        id: "i78h0bobom"
+      }
+    ];
+  }
+
+  handleItemClick() {
```
Comment (ihor7016) [inline src/components/accounts/accounts.js]: do we need to handle click on items?

File: src/components/tags/tags.js [hunk 30]
```diff
@@ -0,0 +1,23 @@
...
+import ngMaterial from "angular-material";
+import template from "./tags.html";
+import { buttonMoreModule } from "../button-more/button-more";
+
+class TagsController {
+  constructor() {
+    this.tags = ["tag1", "tag2", "tag3"];
+  }
+
+  handleItemClick() {
```
Comment (ihor7016) [inline src/components/tags/tags.js]: do we need to handle click on items?

File: src/components/accounts/accounts.html [hunk 31]
```diff
@@ -0,0 +1,13 @@
+<md-list-item ng-click="$ctrl.handleAccountClick($event)">
```
Comment (ihor7016) [inline src/components/accounts/accounts.html]: there is no such handler in controller

File: src/components/tags/tags.html [hunk 32]
```diff
@@ -0,0 +1,13 @@
+<md-list-item ng-click="$ctrl.addTagClick($event)">
```
Comment (ihor7016) [inline src/components/tags/tags.html]: there is no such handler in controller

File: src/components/toolbar/toolbar.js [hunk 33]
```diff
@@ -0,0 +1,25 @@
+import angular from "angular";
+import ngAnimate from "angular-animate";
+import ngAria from "angular-aria";
+import ngMaterial from "angular-material";
+import template from "./toolbar.html";
+
+class ToolbarController {
+  handleMenuClick() {
+    this.onMenuClick();
```
Comment (ihor7016) [inline src/components/toolbar/toolbar.js]: do you need this handler? you can directly call `onMenuClick ` in template

File: src/components/drawer/drawer.js [hunk 34]
```diff
@@ -0,0 +1,30 @@
...
+
+export class DrawerController {
+  constructor($mdSidenav) {
+    this.$mdSidenav = $mdSidenav;
+  }
+  handleAccountClick(event) {
+    console.log("handleAccountClick");
+  }
+
+  addTagClick(event) {
```
Comment (ihor7016) [inline src/components/drawer/drawer.js]: handleAddTagClick

File: src/components/accounts/accounts.html (hunk 31 above)
Comment (ihor7016) [inline src/components/accounts/accounts.html]: probably `handleAddAccountClick `

File: src/components/app/app.scss [hunk 35]
```diff
@@ -0,0 +1,10 @@
+.app {
+  min-height: 100vh;
+  &__container {
+    width: 100%;
+    &-content {
```
Comment (ihor7016) [inline src/components/app/app.scss]: please do not do nesting this way --- it is enough to write `.app__container` and `.app__content`

File: src/components/app/app.js [hunk 36]
```diff
@@ -1,6 +1,26 @@
...
-export const appModule = angular.module("appModule", []).component("app", {
-  template
-}).name;
+export class AppController {
+  constructor() {
+    this.tasks = [];
+    this.drawerOpened = true;
+  }
+
+  handleTaskAdd(task) {
```
Comment (gift-a) [inline src/components/app/app.js]: task add?

File: src/components/drawer/drawer.js [hunk 37]
```diff
@@ -0,0 +1,30 @@
...
+
+export class DrawerController {
+  constructor($mdSidenav) {
+    this.$mdSidenav = $mdSidenav;
+  }
+  handleAddTagClick(event) {
+    console.log("handleAddTagClick");
+  }
+
+  handleAddTagClick(event) {
```
Comment (gift-a) [inline src/components/drawer/drawer.js]: duplicate from 13 row. maybe `account`?

Comment (gift-a) [reply src/components/drawer/drawer.js]: why do you need it here? you have the same handler in `tags.js`

**Review by ihor7016 — APPROVED** [review]

**Pull request merged**